tests/e2e/specs/setup-mc/step-1-accounts.test.js

```diff
@@ -28,110 +45,426 @@ test.describe( 'Merchant who is getting started', () => {
-		// When pending even WPORG will not render yet.
-		// If not mocked will fail and render nothing,
-		// as Jetpack is mocked only on the client-side.
-		await page.route( /\/wc\/gla\/google\/connected\b/, ( route ) =>
-			route.fulfill( {
-				content: 'application/json',
-				headers: { 'Access-Control-Allow-Origin': '*' },
-				body: JSON.stringify( {
-					active: 'no',
-					email: '',
-				} ),
-			} )
-		);
-
-		// Go to the setup page short way - directly via URL.
-		await page.goto(
-			'/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc'
-		);
-
-		// Wait for API calls and the page to render.
-		await page.waitForLoadState( 'networkidle' );
-		page.getByRole( 'heading', { name: 'Set up your accounts' } );
+			// Click the enabled connect button.
+			page.locator(
+				"//button[text()='Connect'][not(@disabled)]"
+			).click();
+			await page.waitForLoadState( 'networkidle' );
+
+			// Expect the user to be redirected
+			await page.waitForURL( baseURL + 'auth_url' );
+		} );
 	} );
 
-	test( 'should see their WPORG email, "Google" title & connect button', async ( {
-		page,
-	} ) => {
-		await expect( page.getByText( 'mail@example.com' ) ).toBeVisible();
+	test.describe( 'Connect Google account', () => {
+		test.beforeAll( async () => {
+			// Mock Jetpack as connected
+			await setUpAccountsPage.mockJetpackConnected(
+				'Test user',
+				'jetpack@example.com'
+			);
 
-		await expect(
-			page.getByText( 'Google', { exact: true } )
-		).toBeVisible();
+			// Mock google as not connected.
+			// When pending even WPORG will not render yet.
+			// If not mocked will fail and render nothing,
+			// as Jetpack is mocked only on the client-side.
+			await setUpAccountsPage.mockGoogleNotConnected();
 
-		expect(
-			page.getByRole( 'button', { name: 'Connect' } ).first()
-		).toBeEnabled();
+			await setUpAccountsPage.goto();
+
+			page.getByRole( 'heading', { name: 'Set up your accounts' } );
+		} );
+
+		test( 'should see their WPORG email, "Google" title & connect button', async () => {
+			const jetpackDescriptionRow =
+				setUpAccountsPage.getJetpackDescriptionRow();
+
+			await expect( jetpackDescriptionRow ).toContainText(
+				'jetpack@example.com'
+			);
+
+			await expect(
+				page.getByText( 'Google', { exact: true } )
+			).toBeVisible();
+
+			await expect(
+				page.getByRole( 'button', { name: 'Connect' } ).first()
+			).toBeEnabled();
+		} );
+
+		test( 'after clicking the "Connect your Google account" button should send an API request to connect Google account, and redirect to the returned URL', async ( {
+			baseURL,
+		} ) => {
+			// Mock google connect.
+			await setUpAccountsPage.mockGoogleConnect(
+				baseURL + 'google_auth'
+			);
+
+			// Click the enabled connect button
+			page.locator(
+				"//button[text()='Connect'][not(@disabled)]"
+			).click();
+			await page.waitForLoadState( 'networkidle' );
+
+			// Expect the user to be redirected
+			await page.waitForURL( baseURL + 'google_auth' );
```

Comment:
Same as in https://github.com/woocommerce/google-listings-and-ads/pull/2081/files#r1315777409

We can consider adding `expect( page.url() ).toMatch( baseURL + 'google_auth'  );` as a valid asseertion